app/models/user.rb

```diff
@@ -3,4 +3,9 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
+
+  validates :first_name, presence: true
+  validates :last_name, presence: true
+  validates :user_name, presence: true, uniqueness: true
+  validates :role, presence: true
```

Comment:
You should add some kind of `inclusion` validation to make sure it's one of your expected roles!